sys/hacl/build.rs

```diff
@@ -370,71 +394,54 @@ struct Platform {
     pmull: bool,
     adv_simd: bool,
     sha256: bool,
+    target: Option<&'static str>,
+    target_arch: String,
+    target_env: String,
+    target_os: String,
 }
 
 fn main() {
     // Get ENV variables
     let home_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
     let home_path = Path::new(&home_dir);
-    let out_dir = env::var("OUT_DIR").unwrap();
-    let out_path = Path::new(&out_dir);
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+    let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
 
     // Check platform support
-    let platform = Platform {
-        simd128: libcrux_platform::simd128_support(),
-        simd256: libcrux_platform::simd256_support(),
-        aes_ni: libcrux_platform::aes_ni_support(),
-        x25519: libcrux_platform::x25519_support(),
-        bmi2_adx_support: libcrux_platform::bmi2_adx_support(),
-        pmull: libcrux_platform::pmull_support(),
-        adv_simd: libcrux_platform::adv_simd_support(),
-        sha256: libcrux_platform::sha256_support(),
+    let platform = if target_arch != "wasm32" {
+        let x86 = target_arch == "x86";
+        Platform {
+            simd128: !x86 && libcrux_platform::simd128_support(),
+            simd256: !x86 && libcrux_platform::simd256_support(),
+            aes_ni: libcrux_platform::aes_ni_support(),
+            x25519: !x86 && libcrux_platform::x25519_support(),
+            bmi2_adx_support: libcrux_platform::bmi2_adx_support(),
+            pmull: libcrux_platform::pmull_support(),
+            adv_simd: libcrux_platform::adv_simd_support(),
+            sha256: libcrux_platform::sha256_support(),
+            target: None,
+            target_arch: target_arch.clone(),
+            target_env: target_env.clone(),
+            target_os: target_os.clone(),
+        }
+    } else {
+        Platform {
+            target_arch: target_arch.clone(),
+            target_env: target_env.clone(),
+            target_os: target_os.clone(),
+            ..Default::default()
+        }
     };
 
     // Set re-run trigger for all of c
     println!("cargo:rerun-if-changed=c");
 
     // Build the C files
-    build(platform, home_path);
-
-    // Set library name to look up
-    const LIB_NAME: &str = "hacl";
+    build(&platform, home_path);
 
     // Generate new bindings.
-    create_bindings(platform, home_path);
-
-    // Link hacl library.
-    const MODE: &str = "static";
-    println!("cargo:rustc-link-lib={}={}", MODE, LIB_NAME);
-    if platform.simd128 {
-        println!("cargo:rustc-link-lib={}={}", MODE, "hacl_128");
```

Comment:
By having the different lib naming, we no longer have to link the additional libraries manually?

Reply:
Oh looks like the comment didn't make it here.
This is unrelated to names. `cc` adds this boilerplate if not disabled.